### ISSUE TITLE
Consideration: call out when Haskell bleeds through

### ISSUE THREAD

**Issue by netaisllc** [elm-guides/elm-for-js#7]
Coming strictly from a JS background, it can be quite problematic to distinguish between **Elm** specifics and **Haskell** concepts that have been borrowed, extended or used.  

---

Case in point:  Today I was puzzled over a particular use of **Just** in StartApp.simple.

My learned process:
1. Scour Elm docs === nothing specific
2. Search elm-discuss === quite difficult to isolate 'Just' from casual uses of the word; no help
3. Assume it's actually part of Haskell, and find [this helpful snippet](http://stackoverflow.com/questions/18808258/what-does-the-just-syntax-mean-in-haskell) on SO.

---

The point being there are (n) cases of this that will stumble up the garden-variety JSer who, like me, knows only that Haskell is another language. (You can see this frequently in elm-discuss.)

Now  there is _no problem_ with this,  given what Elm is and purports to be, but if a catalog can be started, it would be a huge help. 

Or at the very least an admission somehow, somewhere 'upfront' that Elm borrows from Haskell and the learner should be ready to seek _some_ answers outside the Elm world he/she has entered.

**Comment by mgold** [comment]
I disagree that the problem is "you need to seek answers from Haskell". I think the problem is that [the Elm docs](http://package.elm-lang.org/packages/elm-lang/core/2.1.0/Maybe#Maybe) are not sufficiently visible.

In particular, searching for `Just` in the [core package search](http://package.elm-lang.org/packages/elm-lang/core/2.1.0/) doesn't turn up anything. It probably should. Since going to the docs was your first inclination, we need to make that fruitful. I've filed elm-lang/package.elm-lang.org#92 to that effect.

**Comment by jamischarles** [comment]
I agree that long term you shouldn't have to even know that Haskell exists. For the time being Haskell docs are more mature (simply due to being around longer). 

I found myself going through tryhaskell.org to understand Elm better (but was confused once or twice). 

I think the solution here is better Elm docs, and more approachable to JS people who don't even know that Haskell exists. 

I really really want a tryelm.org that is just like tryhaskell.org.

**Comment by mgold** [comment]
Have you seen [elm-lang.org/try](http://elm-lang.org/try)? Granted it's a file-like document, not a repl.

Haskell docs are more mature, but they're also more confusing, both because they reference language features Elm doesn't have (like type classes) and because the people who wrote them have PhDs in math.

**Comment by jamischarles** [comment]
> and because the people who wrote them have PhDs in math

Lol. 

I've tried the link above. It's great for seeing a simple program. 

What it's lacking is a tutorial that teaches you syntax as you go

**Comment by jamischarles** [comment]
Most languages have a very simple "try" page that teaches you the syntax. I've used it for Haskell, Clojure and Erlang. It's super fun and is the easiest way I have seen to significantly lower the barrier to entry.

**Comment by netaisllc** [comment]
@mgold. I agree with your disagreement. My point is that the current state of the Elm docs leaves someone new to Elm and Haskell no choice but to look outside the Elm world.

@mgold, @jamischarles As far as Elm-try goes, it's awesome. But the examples - though very worthwhile - "leave a lot of value on the table" because they mostly lack any comments or explanation. They are very useful to show **how** but not helpful about the **why**.

As far as a dedicated entry-point to the Elm for people coming from the traditional webapp stack (if there is such a thing...), I thinking is: **it is absolutely needed**.  Part of the challenged Elm faces is that it is/could be attractive to a wide range of devs with a wide variety of backgrounds.

Having a "this is for you" type of entry point is a proven technique to get people of that "ilk" engaged.

I'd be one to contribute to that effort.

**Comment by mgold** [comment]
I agree with what you're saying. If you want to improve the examples, [have at it](https://github.com/elm-guides/elm-lang.org/tree/master/src/examples) and send a PR. Many times an example fails to generalize: "okay, I get how this works, but why is it not any one of these other things? How would I do something slightly different?". Comments can help with this.
